notification-rule-config-service-impl/src/main/java/org/hypertrace/notification/config/service/NotificationRuleFilteredStore.java

```diff
@@ -52,4 +54,16 @@ protected Value buildValueFromData(NotificationRule object) {
   protected String getContextFromData(NotificationRule object) {
     return object.getId();
   }
+
+  @Override
+  protected Optional<NotificationRule> filterConfigData(
+      NotificationRule data, NotificationRuleFilter filter) {
+    return Optional.of(data)
+        .filter(
+            notificationRule ->
+                filter
```

Comment:
Check a condition for hasFilter.
We don't want to apply any filter if filter is not set or list is empty.